includes/Modules/AdSense/Tag_Guard.php

```diff
@@ -30,6 +30,12 @@ class Tag_Guard extends Module_Tag_Guard {
 	 */
 	public function can_activate() {
 		$settings = $this->settings->get();
+
+		// For web stories, the tag must only be rendered if a story-specific ad unit is provided.
+		if ( is_singular( 'web-story' ) && ! $settings['webStoriesAdUnit'] ) {
```

Comment:
Nitpick but let's check whether the webStoriesAdUnit setting is empty:

```suggestion
		if ( is_singular( 'web-story' ) && empty( $settings['webStoriesAdUnit'] ) ) {
```